### ISSUE TITLE
Flow-Typed Support?

### ISSUE THREAD

**Issue by todesignandconquer** [FormidableLabs/redux-little-router#148]
I noticed that you're using Flow but I was wondering if there were any plans to contribute your definitions to Flow Typed?

**Comment by DrummerHead** [comment]
I was wondering the same :) and thanks for the project, what a pleasure!

**Comment by DrummerHead** [comment]
I'm using this limited library definition, it's incomplete but it can be a nice start for making a complete one :)

```
declare module 'redux-little-router' {
  declare type HistoryLocation = {
    pathname: string,
    search?: string,
    hash?: string,
    state?: Object,
    key?: string
  };

  declare type Location = HistoryLocation & {
    basename?: string,
    options?: LocationOptions,
    params?: Params,
    previous?: Location,
    query?: Query,
    result?: Object
  };

  declare type BrowserRouterArgs = {
    routes: Object,
    basename?: string,
    getLocation?: () => Location,
    passRouterStateToReducer?: bool
  };

  declare type LocationAction = {
    type: 'ROUTER_LOCATION_CHANGED' | 'ROUTER_PUSH' | 'ROUTER_REPLACE',
    payload: Location
  };

  declare function Reducer(Location, LocationAction): Location;
  declare function ReducerGen(Location): Reducer;

  declare export function routerForBrowser(BrowserRouterArgs): {
    reducer: ReducerGen,
    middleware: any,
    enhancer: any,
  };
//  declare export function routerForExpress(any): any;
//  declare export function routerForHapi(any): any;
  declare export function initializeCurrentLocation(any): any;
//
//  declare export function provideRouter(any): any;
  declare export function RouterProvider(any): any;
//  declare export function Link(any): any;
//  declare export PersistentQueryLink: any;
//  declare export Fragment: any;
//
  declare export function push(any): any;
//  declare export replace: any;
//  declare export go: any;
//  declare export goBack: any;
//  declare export goForward: any;
//
//  declare export LOCATION_CHANGED: any;
//  declare export PUSH: any;
//  declare export REPLACE: any;
//  declare export GO: any;
//  declare export GO_FORWARD: any;
//  declare export GO_BACK: any;
}
```

**Comment by tptee** [comment]
Hey all! Definitely want to make this happen. Will keep you posted!

**Comment by tptee** [comment]
#257 exports the previously internal Flow types! Since we won't likely have the bandwidth to test against Flow version ranges. this'll remain in the `npm` package instead of `flow-typed` and will be guaranteed to work only with the version of `flow` we depend on (currently `0.64`).

**Comment by tptee** [comment]
Released in `v14.3.0`
